Cleaning the contact_map

hengma1001 opened the pull request:
1. Moved the contact_size definition before the h5 file path where the variable is used 
2. Saved the whole contact_matrices to h5 file instead of doing it line by line 
3. Moved file saving out of trajectory loop

hengma1001 committed:
Cleaning up the contact map

cleaning up on minor bugs on save path and others

hengma1001 committed:
Fixed the cm_table shape

yngtodd commented on molecules/sim/md_analysis/contact_maps.py:
```diff
@@ -15,25 +15,25 @@ def contact_maps_from_traj(xyz_file, traj_file, cm_table_file, contact_cutoff=8.
+        outfile.close() 
+        return outpath
 
     return contact_matrices
```
What would happen if you want to both write to disk and return `contact_matrices` for other parts of the program?

yngtodd replied:
Cool, that looks good! 👍

hengma1001 committed:
Trimmed contact_maps.py

1. Removed cm_table_file input
2. Merged the savepath and filename in input
3. Added return absolute savefile if the contact map stored in file

Commit:
add function convert upper triangle to full contact matrix

yngtodd commented on molecules/utils/matrix_op.py:
```diff
@@ -0,0 +1,12 @@
+    cm_full.T[iu1] = cm0 
+    np.fill_diagonal(cm_full, 1)
+    
+    return cm_full
```
Nice. You are awesome.

hengma1001 commented:
Let me add that also. Lol


On Oct 25, 2018, at 4:19 PM, Todd Young <notifications@github.com<mailto:notifications@github.com>> wrote:


@yngtodd commented on this pull request.

________________________________

In molecules/utils/matrix_op.py<https://github.com/yngtodd/molecules/pull/6#discussion_r228320169>:

> @@ -0,0 +1,12 @@
+import numpy as np
+
+def tria_to_full(cm0):
+    num_res = int(np.ceil((len(cm0) * 2) ** 0.5))
+    iu1 = np.triu_indices(num_res, 1)
+
+    cm_full = np.zeros((num_res, num_res))
+    cm_full[iu1] = cm0
+    cm_full.T[iu1] = cm0
+    np.fill_diagonal(cm_full, 1)
+
+    return cm_full


Nice. You are awesome.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/yngtodd/molecules/pull/6#pullrequestreview-168568416>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AksEdn29CKaoEPRaQ7hDxpjm4SyXcI08ks5uohzBgaJpZM4X4RDi>.

hengma1001 committed:
Updated contact_maps.py

1. Removed 'return savefile', always return contact_matrices
2. Removed num_ca and contact_size as no longer validate, pointed out by Chris.

Commit:
change the name from tria to triu indicating upper triangle

Commit:
Merge branch 'master' of https://github.com/hengma1001/molecules

getting updates from Todd.

Commit:
read h5 file

Commit:
read h5 file

yngtodd commented:
Cool! I'm going to go ahead and merge this one, and we can open up a new pull request if there is anything else.
